pkg/kubelet/kubelet_network.go

```diff
@@ -124,6 +124,9 @@ func (kl *Kubelet) providerRequiresNetworkingConfiguration() bool {
 	if kl.cloud == nil || kl.cloud.ProviderName() != "gce" || kl.flannelExperimentalOverlay {
 		return false
 	}
+	if kl.networkPlugin != nil && !kl.networkPlugin.Capabilities().Has(network.NET_PLUGIN_CAPABILITY_CLOUD_BASED) {
```

Comment:
What did you mean by "the network provider is not validating that nodes are ready"?

The current behaviour is that code inside kubelet marks the node as not-ready, and code inside the controller-manager then marks it as ready when it thinks it ought to be, without anyone validating anything.